### PR TITLE
Update scaladoc api mappings

### PR DESCRIPTION
**Pull request by regadas**
This adds a little plugin that removes the need for all the cruft around `apiMappings`

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -19,6 +19,7 @@ addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.3.5")
 addSbtPlugin("org.scalameta" % "sbt-mdoc" % "2.1.5")
 addSbtPlugin("com.lightbend.sbt" % "sbt-java-formatter" % "0.5.1")
 addSbtPlugin("com.github.cb372" % "sbt-explicit-dependencies" % "0.2.12")
+addSbtPlugin("com.thoughtworks.sbt-api-mappings" % "sbt-api-mappings" % "3.0.0")
```
Comment (nevillelyh) [inline project/plugins.sbt]: Do they have hard coded mapping for Dataflow, BigQuery JavaDocs etc.?
TBH those are nice to have I don't mind not having the link & keep this simple.

Comment (regadas) [reply project/plugins.sbt]: The plugin resolves the javadoc.io links for them ... so when we have `PCollection` in a scaladoc ... users can actually jump to the reespective javadoc

Comment (nevillelyh) [reply project/plugins.sbt]: I'm just curious how does it know the URL prefix for say Beam JavaDoc?

Comment (nevillelyh) [reply project/plugins.sbt]: https://github.com/ThoughtWorksInc/sbt-api-mappings/tree/3.0.x/src/main/scala/com/thoughtworks/sbtApiMappings
So looks like a bunch of hard coded ones including JDK lib, spark, play, but that's about it. 🤷‍♂

Comment (regadas) [reply project/plugins.sbt]: for most of the deps will use the sonatype mapping rule.

**Pull request merged**